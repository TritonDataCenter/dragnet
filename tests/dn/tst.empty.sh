#!/bin/bash

#
# tst.empty.sh: tests basic "scan" operations on empty files
#

set -o errexit
. $(dirname $0)/common.sh

tmpfile="/var/tmp/$(basename $0).$$"
echo "using tmpfile \"$tmpfile\"" >&2

function scan
{
	echo "# dn scan" "$@"
	dn scan "$@" /dev/null 2>&1
	echo

	echo "# dn scan --points" "$@"
	dn scan --points "$@" /dev/null 2>&1 | sort -d
	echo
}

function query
{
	echo "# dn query" "$@"
	dn query "$@" $tmpfile 2>&1
}

scan --counters
scan -b timestamp
scan -b timestamp[aggr=quantize]
scan -b timestamp[aggr=quantize],req.method
scan -f '{ "eq": [ "audit", true ] }' -b timestamp[aggr=quantize],req.method
scan --counters -f '{ "eq": [ "audit", true ] }'

echo "creating index" >&2
dn index /dev/null $tmpfile
query --counters

echo "creating index" >&2
dn index -c req.method,latency[aggr=quantize] /dev/null $tmpfile
query --counters
query -f '{ "eq": [ "req.method", "GET" ] }'
query -b req.method
query -b latency
query --counters -b latency
