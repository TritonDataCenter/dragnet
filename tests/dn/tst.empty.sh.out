# dn scan --counters
VALUE
    0
Aggregator         noutputs:           1
Flattener          ninputs:            1
Flattener          noutputs:           1

# dn scan --points --counters
Aggregator         noutputs:           1
{"fields":{},"value":0}

# dn scan -b timestamp

# dn scan --points -b timestamp

# dn scan -b timestamp[aggr=quantize]

# dn scan --points -b timestamp[aggr=quantize]

# dn scan -b timestamp[aggr=quantize],req.method

# dn scan --points -b timestamp[aggr=quantize],req.method

# dn scan -f { "eq": [ "audit", true ] } -b timestamp[aggr=quantize],req.method

# dn scan --points -f { "eq": [ "audit", true ] } -b timestamp[aggr=quantize],req.method

# dn scan --counters -f { "eq": [ "audit", true ] }
VALUE
    0
Aggregator         noutputs:           1
Flattener          ninputs:            1
Flattener          noutputs:           1

# dn scan --points --counters -f { "eq": [ "audit", true ] }
Aggregator         noutputs:           1
{"fields":{},"value":0}

# dn query --counters
VALUE
    0
Index Aggregator   ninputs:            1
Index Aggregator   noutputs:           1
Query Passthrough  ninputs:            1
Query Passthrough  noutputs:           1
Flattener          ninputs:            1
Flattener          noutputs:           1
# dn query --counters
VALUE
    0
Index Aggregator   ninputs:            1
Index Aggregator   noutputs:           1
Query Passthrough  ninputs:            1
Query Passthrough  noutputs:           1
Flattener          ninputs:            1
Flattener          noutputs:           1
# dn query -f { "eq": [ "req.method", "GET" ] }
VALUE
    0
# dn query -b req.method
# dn query -b latency
# dn query --counters -b latency
Flattener          noutputs:           1
